compiler/modules/types/core.bal

```diff
@@ -960,7 +960,7 @@ function bddMappingAtomicType(Env env, Bdd bdd, MappingAtomicType top) returns M
 // for when T is a subtype of mapping, and K is either `string` or a singleton string.
 // This is what Castagna calls projection.
 // We will extend this to allow `key` to be a SemType, which will turn into a StringSubtype.
-public function mappingMemberType(Context cx, SemType t, string? k = ()) returns SemType {
+public function mappingMemberType(Context cx, SemType t, string|SemType? k = ()) returns SemType {
```

Comment:
Type of `k` should just be SemType.

You need to fix `mappingMemberRequired` also (also change `k` to `SemType`)

Reply:
**all** string to represent the present `nil`?

Reply:
@jclark what should `mappingMemberRequired` do when `k` contains more than one string singleton?

Reply:
Made it so that if all singleton strings in the string subtype are field names then to return `true` from `mappingMemberRequired`